test/unit/googleError.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import {describe, it} from 'mocha';
+import * as fs from 'fs';
+import * as path from 'path';
+import {GoogleErrorDecoder} from '../../src/googleError';
+
+interface MyObj {
+  type: string;
+  value: protobuf.Message<{}>;
+}
+
+describe('gRPC-google error decoding', () => {
+  it('decodes error', () => {
+    // example of an actual google.rpc.Status error message returned by Language API
+    const fixtureName = path.resolve(
+      __dirname,
+      '..',
+      'fixtures',
+      'badRequest.bin'
+    );
+    const errorBin = fs.readFileSync(fixtureName);
+    const expectedError = {
+      fieldViolations: [
+        {
+          field: 'document.language',
+          description: 'The document language is not valid',
+        },
+      ],
+    };
+    const decoder = new GoogleErrorDecoder();
+
+    const decodedError = decoder.decodeRpcStatusDetails([errorBin])[0];
+
+    // nested error messages have different types so we can't use deepStrictEqual here
+    assert.strictEqual(
+      JSON.stringify(decodedError),
+      JSON.stringify(expectedError)
+    );
+  });
+
+  it('decodes multiple errors', () => {
+    // example of when there are multiple errors available to be decoded
+    const bufferArr = [] as Buffer[];
+    let decodedErrorArr = [] as protobuf.Message<{}>[];
+    const expectedErrorArr = [] as protobuf.Message<{}>[];
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const protobuf = require('protobufjs');
+    const decoder = new GoogleErrorDecoder();
+
+    const fixtureName = path.resolve(
+      __dirname,
+      '..',
+      'fixtures',
+      'multipleErrors.json'
+    );
+    const protos_path = path.resolve(
+      __dirname,
+      '..',
+      '..',
+      'protos',
+      'google',
+      'rpc'
+    );
+    fs.readFile(fixtureName, 'utf8', (err, data) => {
+      const objs = JSON.parse(data) as MyObj[];
+      for (const obj of objs) {
+        protobuf.load(
+          [protos_path + '/error_details.proto', protos_path + '/status.proto'],
```

Comment:
I believe you could do this `protobuf.load` just once on top of the file.  It's OK to do `loadSync` to get rid of the callback.  So it would be `const root = protobuf.loadSync([path.join(protos_path, 'error_details.proto'), path.join(protos_path, 'status.proto')`  (please use `path.join` instead of concatenation).